### DISCUSSION TITLE
Issues with the Sous Vide node

### DISCUSSION THREAD

**Discussion by Rodriweiss** [andrei-tatar/node-red-contrib-smartnora#260] [Q&A]
Hello. I've been trying to create a cooking thermometer that warns me when the liquid reaches a certain temperature using a nodemcu. While not the most elegant of solutions, I got it to work by comparing the values in the ambient and set temperatures and forwarding the message to a chromecast node when the condition is true

What I found odd is the behavior of the device in the Google Home app (I know a new version is being rolled but I don't have it yet). By which I mean, the thermometer sensor is able to feed the current temperature correctly to the node and I'm able to set the desired temperature by voice with "Google set the sous vide to n". What I couldn't find was a way to turn the device on and off by voice -the assistant says it can't find the device, even though it's right there and I can flick the switch manually in the app.

If I create a custom command on the Google Home app to turn the device on and off it works but the assistant will no longer accept commands to set the temperature until I delete it. This is so odd.

**Comment by andrei-tatar** [comment]
@Rodriweiss did you also enable the on/off trait on the sous vide? I admit, I haven't tried it myself.

**Comment by Rodriweiss** [comment]
Yes, the on and off switch only appears in the google home app with the trait ticked on. Without the trait it just shows up as a generic sensor but otherwise still works with voice the command to set the temperature. Just to be sure I set up the nodemcu to send and receive mqtt payloads with on and off messages but otherwise do nothing else, still doesn't work.

I'm guessing this is an issue on google's end? It should be able to turn the device on and off by voice if I can do it manually in the app, right?

**Reply by andrei-tatar** [reply]
Yup, if it works in Google Home, should work in the assistant as well. 
Maybe also try assigning it to a Home/Room, in a Google Home. That might fix it as well (if not already assigned).

**Comment by Rodriweiss** [comment]
It is already assigned to my kitchen. As I said, if I set up a custom command in the home app, I can turn it on and off, however it will no longer accept temperature commands. Hopefully this is something they have fixed in the new app...

**Reply by andrei-tatar** [reply]
Can you export a test flow you're using. I already have the new app, I can test it.

**Reply by Rodriweiss** [reply]
Sure

[{"id":"1bebd9319c2eed85","type":"ui_gauge","z":"f7a1060f7191035d","name":"Kitchen Thermometer","group":"5b8c9e45308bc8d4","order":0,"width":0,"height":0,"gtype":"gage","title":"Kitchen Thermometer","label":"Cº","format":"{{value}}","min":0,"max":"150","colors":["#00b500","#e6e600","#ca3838"],"seg1":"","seg2":"","x":340,"y":1600,"wires":[]},{"id":"bcf396d1f693a9f6","type":"noraf-sousvide","z":"f7a1060f7191035d","devicename":"Kitchen Helper","roomhint":"Kitchen","name":"Kitchen Helper","onOffSupported":false,"onOffMode":"cq","temperatureSupported":true,"temperatureMode":"cq","temperatureUnit":"C","temperatureStepSize":1,"temperatureRangeMin":"0","temperatureRangeMax":"120","startStopSupported":false,"startStopPausable":false,"timerSupported":false,"timerMaxLimitSeconds":3600,"timerMode":"cq","topic":"kitchen/helper/pub","passthru":true,"nora":"b2d5f94a6cf38ae8","x":460,"y":1700,"wires":[["04ace71a372ce030"]]},{"id":"affbf2181642456c","type":"debug","z":"f7a1060f7191035d","name":"","active":true,"tosidebar":true,"console":false,"tostatus":false,"complete":"false","statusVal":"","statusType":"auto","x":970,"y":1660,"wires":[]},{"id":"5b41ea71538e6ce7","type":"delay","z":"f7a1060f7191035d","name":"Delay","pauseType":"rate","timeout":"5","timeoutUnits":"seconds","rate":"1","nbRateUnits":"1","rateUnits":"minute","randomFirst":"1","randomLast":"5","randomUnits":"seconds","drop":false,"allowrate":false,"x":790,"y":1700,"wires":[["affbf2181642456c","6001bb65dcb028da"]]},{"id":"6001bb65dcb028da","type":"cast-to-client","z":"f7a1060f7191035d","name":"Message","url":"","contentType":"","message":"The desired temperature has been reached","language":"en","ip":"192.168.100.135","port":"","volume":"","x":960,"y":1700,"wires":[[]]},{"id":"f7bf43af4844402e","type":"mqtt in","z":"f7a1060f7191035d","name":"Kitchen Thermometer","topic":"kitchen/temperature/pub","qos":"2","datatype":"json","broker":"c45e2cc547ba88b2","nl":false,"rap":true,"rh":0,"x":100,"y":1700,"wires":[["aa50f07310d8e618","1bebd9319c2eed85"]]},{"id":"aa50f07310d8e618","type":"function","z":"f7a1060f7191035d","name":"","func":"tempTherm = msg.payload;\n\n\nmsg.payload = {\n    thermostatTemperatureAmbient: parseInt(tempTherm),\n\n}\nreturn msg","outputs":1,"noerr":0,"initialize":"","finalize":"","libs":[],"x":300,"y":1700,"wires":[["bcf396d1f693a9f6"]]},{"id":"04ace71a372ce030","type":"function","z":"f7a1060f7191035d","name":"","func":"\n\nlet temp1 = parseInt(msg.payload.thermostatTemperatureAmbient);\nlet temp2 = parseInt(msg.payload.thermostatTemperatureSetpoint);\n\nif (!temp2) return {\n}\nelse {\nif (temp1 == temp2){\n    \n    msg.payload = \"The desired temperature has been reached\"\n    \n    return msg}\n}","outputs":1,"noerr":0,"initialize":"","finalize":"","libs":[],"x":640,"y":1700,"wires":[["5b41ea71538e6ce7"]]},{"id":"5b8c9e45308bc8d4","type":"ui_group","name":"Sensor","tab":"bd1610225d88de3a","order":2,"disp":true,"width":"6","collapse":false},{"id":"b2d5f94a6cf38ae8","type":"noraf-config","name":"Nora Devices","group":"Devices","twofactor":"off","twofactorpin":"","localexecution":true,"structure":"","storeStateInContext":false,"disableValidationErrors":false,"sendDeviceNameAndLocation":false},{"id":"c45e2cc547ba88b2","type":"mqtt-broker","name":"localhost:1883","broker":"localhost","port":"1883","clientid":"","usetls":false,"protocolVersion":"3","keepalive":"60","cleansession":true,"birthTopic":"","birthQos":"0","birthPayload":"","birthMsg":{},"closeTopic":"","closeQos":"0","closePayload":"","closeMsg":{},"willTopic":"","willQos":"0","willPayload":"","willMsg":{},"sessionExpiry":""},{"id":"bd1610225d88de3a","type":"ui_tab","name":"Room","icon":"dashboard","order":1,"disabled":false,"hidden":false}]

**Reply by andrei-tatar** [reply]
@Rodriweiss in the flow you exported, it seems like the `On/Off` trait is not enabled:
![image](https://github.com/andrei-tatar/node-red-contrib-smartnora/assets/6473183/f5601b7f-0e9b-4d75-8829-9eff1b002241)

**Reply by Rodriweiss** [reply]
I must have ticked it off since it wasn't working anyway, I had it on before
